Unable to connect Presto in v0.34.0

**Describe the bug**
We are not able to connect to PrestoDB from Metabase v0.34.0. We were able to connect it in v0.33.6. We get following error in UI (Removed Hostname and Username from image):
![image](https://user-images.githubusercontent.com/55583247/71504775-b54fb680-28a0-11ea-923a-0491338b1af3.png)


**Logs**
Please find below server logs:
```
12-26 16:31:47 [1mERROR driver.util[0m :: Database connection error

16:31:47
java.lang.AssertionError: Assert failed: ((some-fn keyword? string?) driver)

16:31:47
at metabase.driver$the_driver.invokeStatic(driver.clj:97)

16:31:47
at metabase.driver$the_driver.invoke(driver.clj:97)

16:31:47
at metabase.driver$the_initialized_driver.invokeStatic(driver.clj:145)

16:31:47
at metabase.driver$the_initialized_driver.invoke(driver.clj:142)

16:31:47
at metabase.driver$dispatch_on_initialized_driver.invokeStatic(driver.clj:154)

16:31:47
at metabase.driver$dispatch_on_initialized_driver.doInvoke(driver.clj:149)

16:31:47
at clojure.lang.RestFn.invoke(RestFn.java:410)

16:31:47
at metabase.driver$fn__21027$fn__21028.invoke(driver.clj:399)

16:31:47
at clojure.lang.MultiFn.invoke(MultiFn.java:233)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invokeStatic(timezone.clj:45)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invoke(timezone.clj:38)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invokeStatic(timezone.clj:42)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invoke(timezone.clj:38)

16:31:47
at metabase.driver.presto$details__GT_request.invokeStatic(presto.clj:54)

16:31:47
at metabase.driver.presto$details__GT_request.invoke(presto.clj:49)

16:31:47
at metabase.driver.presto$execute_presto_query_BANG_$fn__318.invoke(presto.clj:101)

16:31:47
at metabase.util.ssh$with_ssh_tunnel_STAR_.invokeStatic(ssh.clj:106)

16:31:47
at metabase.util.ssh$with_ssh_tunnel_STAR_.invoke(ssh.clj:91)

16:31:47
at metabase.driver.presto$execute_presto_query_BANG_.invokeStatic(presto.clj:98)

16:31:47
at metabase.driver.presto$execute_presto_query_BANG_.invoke(presto.clj:94)

16:31:47
at metabase.driver.presto$eval336$can_connect_QMARK___335__347$fn__351.invoke(presto.clj:138)

16:31:47
at metabase.driver.presto$eval336$can_connect_QMARK___335__347.invoke(presto.clj:136)

16:31:47
at clojure.lang.AFn.applyToHelper(AFn.java:156)

16:31:47
at clojure.lang.AFn.applyTo(AFn.java:144)

16:31:47
at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)

16:31:47
at clojure.lang.RestFn.invoke(RestFn.java:421)

16:31:47
at clojure.lang.MultiFn.invoke(MultiFn.java:234)

16:31:47
at metabase.driver.util$can_connect_with_details_QMARK_$fn__21110.invoke(util.clj:32)

16:31:47
at metabase.util$do_with_timeout$fn__13131.invoke(util.clj:302)

16:31:47
at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)

16:31:47
at clojure.lang.AFn.call(AFn.java:18)

16:31:47
at java.base/java.util.concurrent.FutureTask.run(Unknown Source)

16:31:47
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)

16:31:47
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

16:31:47
at java.base/java.lang.Thread.run(Unknown Source)

16:31:47
12-26 16:31:47 [1mERROR driver.util[0m :: Database connection error

16:31:47
java.lang.AssertionError: Assert failed: ((some-fn keyword? string?) driver)

16:31:47
at metabase.driver$the_driver.invokeStatic(driver.clj:97)

16:31:47
at metabase.driver$the_driver.invoke(driver.clj:97)

16:31:47
at metabase.driver$the_initialized_driver.invokeStatic(driver.clj:145)

16:31:47
at metabase.driver$the_initialized_driver.invoke(driver.clj:142)

16:31:47
at metabase.driver$dispatch_on_initialized_driver.invokeStatic(driver.clj:154)

16:31:47
at metabase.driver$dispatch_on_initialized_driver.doInvoke(driver.clj:149)

16:31:47
at clojure.lang.RestFn.invoke(RestFn.java:410)

16:31:47
at metabase.driver$fn__21027$fn__21028.invoke(driver.clj:399)

16:31:47
at clojure.lang.MultiFn.invoke(MultiFn.java:233)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invokeStatic(timezone.clj:45)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invoke(timezone.clj:38)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invokeStatic(timezone.clj:42)

16:31:47
at metabase.query_processor.timezone$report_timezone_id_if_supported.invoke(timezone.clj:38)

16:31:47
at metabase.driver.presto$details__GT_request.invokeStatic(presto.clj:54)

16:31:47
at metabase.driver.presto$details__GT_request.invoke(presto.clj:49)

16:31:47
at metabase.driver.presto$execute_presto_query_BANG_$fn__318.invoke(presto.clj:101)

16:31:47
at metabase.util.ssh$with_ssh_tunnel_STAR_.invokeStatic(ssh.clj:106)

16:31:47
at metabase.util.ssh$with_ssh_tunnel_STAR_.invoke(ssh.clj:91)

16:31:47
at metabase.driver.presto$execute_presto_query_BANG_.invokeStatic(presto.clj:98)

16:31:47
at metabase.driver.presto$execute_presto_query_BANG_.invoke(presto.clj:94)

16:31:47
at metabase.driver.presto$eval336$can_connect_QMARK___335__347$fn__351.invoke(presto.clj:138)

16:31:47
at metabase.driver.presto$eval336$can_connect_QMARK___335__347.invoke(presto.clj:136)

16:31:47
at clojure.lang.AFn.applyToHelper(AFn.java:156)

16:31:47
at clojure.lang.AFn.applyTo(AFn.java:144)

16:31:47
at clojure.lang.AFunction$1.doInvoke(AFunction.java:31)

16:31:47
at clojure.lang.RestFn.invoke(RestFn.java:421)

16:31:47
at clojure.lang.MultiFn.invoke(MultiFn.java:234)

16:31:47
at metabase.driver.util$can_connect_with_details_QMARK_$fn__21110.invoke(util.clj:32)

16:31:47
at metabase.util$do_with_timeout$fn__13131.invoke(util.clj:302)

16:31:47
at clojure.core$binding_conveyor_fn$fn__5754.invoke(core.clj:2030)

16:31:47
at clojure.lang.AFn.call(AFn.java:18)

16:31:47
at java.base/java.util.concurrent.FutureTask.run(Unknown Source)

16:31:47
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)

16:31:47
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)

16:31:47
at java.base/java.lang.Thread.run(Unknown Source)

16:31:47
12-26 16:31:47 [1mDEBUG middleware.log[0m :: [32mPOST /api/database/validate 200 7.5 ms (0 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (71 total active threads) Queries in flight: 0[0m

16:31:57
12-26 16:31:57 [1mDEBUG middleware.log[0m :: [32mGET /api/public/card/61353e2d-f20b-4a05-9a60-a9d69b8e9f9d 200 2.2 ms (1 DB calls) Jetty threads: 3/50 (4 idle, 0 queued) (71 total active threads) Queries in flight: 0[0m

16:31:57
12-26 16:31:57 [1mDEBUG middleware.log[0m :: [31mGET /api/user/current 401 168.5 µs (0 DB calls)

16:31:57
"Unauthenticated"

16:31:57
[0m

16:32:01
12-26 16:32:01 [1mDEBUG middleware.log[0m :: [31mGET /api/user/current 401 153.7 µs (0 DB calls)

16:32:01
"Unauthenticated"
```


**Expected behavior**
Should get connected

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Information about your Metabase Installation:**

```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Safari/605.1.15",
    "vendor": "Apple Computer, Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.3+7",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.3",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.3+7",
    "os.name": "Linux",
    "os.version": "4.4.0-1083-aws",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "redshift",
      "mongo",
      "presto"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2019-12-19",
      "tag": "v0.34.0",
      "branch": "release-0.34.x",
      "hash": "1335215"
    },
    "settings": {
      "report-timezone": "Asia/Calcutta"
    }
  }
}
```

**Severity**
It is VERY SEVERE issue, blocking our main PrestoDB metabase usage.

**Additional context**
Something from v0.34.0 got this up. Please fix it asap.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -81,9 +81,6 @@ executors:
     working_directory: /home/circleci/metabase/metabase/
     docker:
       - image: circleci/clojure:lein-2.8.1
-        environment:
-          MB_PRESTO_TEST_HOST: localhost
-          MB_PRESTO_TEST_PORT: 8080
       - image: metabase/presto-mb-ci
         environment:
           JAVA_TOOL_OPTIONS: "-Xmx2g"

--- a/modules/drivers/presto/src/metabase/driver/presto.clj
+++ b/modules/drivers/presto/src/metabase/driver/presto.clj
@@ -51,7 +51,7 @@
                            "X-Presto-User"   user}
                           (when catalog
                             {"X-Presto-Catalog" catalog})
-                          (when-let [report-timezone (qp.timezone/report-timezone-id-if-supported)]
+                          (when-let [report-timezone (qp.timezone/report-timezone-id-if-supported :presto)]
                             {"X-Presto-Time-Zone" report-timezone}))}
          (when password
            {:basic-auth [user password]})))

--- a/modules/drivers/presto/test/metabase/driver/presto_test.clj
+++ b/modules/drivers/presto/test/metabase/driver/presto_test.clj
@@ -6,7 +6,6 @@
             [metabase
              [driver :as driver]
              [query-processor :as qp]
-             [query-processor-test :as qp.test]
              [test :as mt]]
             [metabase.db.metadata-queries :as metadata-queries]
             [metabase.driver
@@ -16,13 +15,10 @@
             [metabase.models
              [field :refer [Field]]
              [table :as table :refer [Table]]]
-            [metabase.query-processor.test-util :as qp.test-util]
-            [metabase.test
-             [data :as data]
-             [util :as tu]]
-            [metabase.test.data.datasets :as datasets]
             [metabase.test.util.log :as tu.log]
             [toucan.db :as db]))
+
+(use-fixtures :once (mt/initialize-if-needed! :db))
 
 (deftest details->uri-test
   (is (= "http://localhost:8080/"
@@ -48,11 +44,15 @@
                       "X-Presto-Catalog"   "test_data"
                       "X-Presto-Time-Zone" "America/Toronto"}}
            (mt/with-report-timezone-id "America/Toronto"
-             (#'presto/details->request {:user "user", :catalog "test_data"}))))))
+             (#'presto/details->request {:user "user", :catalog "test_data"})))))
+  (testing "details->request should still work if driver is unbound, e.g. when connecting a DB (#11598)"
+    (is (= {:headers {"X-Presto-Source" "metabase"
+                      "X-Presto-User"   nil}}
+           (#'presto/details->request {})))))
 
 (deftest parse-results-test
   (driver/with-driver :presto
-    (qp.test-util/with-everything-store
+    (mt/with-everything-store
       (is (= [["2017-04-03"
                (t/zoned-date-time "2017-04-03T10:19:17.417-04:00[America/Toronto]")
                (t/zoned-date-time "2017-04-03T10:19:17.417Z[UTC]")
@@ -68,12 +68,12 @@
             [[0, false, "", nil]]))))))
 
 (deftest describe-database-test
-  (datasets/test-driver :presto
+  (mt/test-driver :presto
     (is (= {:tables #{{:name "test_data_categories" :schema "default"}
                       {:name "test_data_venues" :schema "default"}
                       {:name "test_data_checkins" :schema "default"}
                       {:name "test_data_users" :schema "default"}}}
-           (-> (driver/describe-database :presto (data/db))
+           (-> (driver/describe-database :presto (mt/db))
                (update :tables (comp set (partial filter (comp #{"test_data_categories"
                                                                  "test_data_venues"
                                                                  "test_data_checkins"
@@ -81,7 +81,7 @@
                                                                :name)))))))))
 
 (deftest describe-table-test
-  (datasets/test-driver :presto
+  (mt/test-driver :presto
     (is (= {:name   "test_data_venues"
             :schema "default"
             :fields #{{:name          "name",
@@ -102,17 +102,17 @@
                       {:name          "id"
                        :database-type "integer"
                        :base-type     :type/Integer}}}
-           (driver/describe-table :presto (data/db) (db/select-one 'Table :id (data/id :venues)))))))
+           (driver/describe-table :presto (mt/db) (db/select-one 'Table :id (mt/id :venues)))))))
 
-;;; TABLE-ROWS-SAMPLE
-(datasets/expect-with-driver :presto
-  [["Red Medicine"]
-   ["Stout Burgers & Beers"]
-   ["The Apple Pan"]
-   ["Wurstküche"]
-   ["Brite Spot Family Restaurant"]]
-  (take 5 (metadata-queries/table-rows-sample (Table (data/id :venues))
-            [(Field (data/id :venues :name))])))
+(deftest table-rows-sample-test
+  (mt/test-driver :prestor
+    (is (= [["Red Medicine"]
+            ["Stout Burgers & Beers"]
+            ["The Apple Pan"]
+            ["Wurstküche"]
+            ["Brite Spot Family Restaurant"]]
+           (take 5 (metadata-queries/table-rows-sample (Table (mt/id :venues))
+                     [(Field (mt/id :venues :name))]))))))
 
 
 ;;; APPLY-PAGE
@@ -150,19 +150,20 @@
     (catch Exception e
       (.getMessage e))))
 
-(datasets/expect-with-driver :presto
-  "UTC"
-  (tu/db-timezone-id))
+(deftest db-default-timezone-test
+  (mt/test-driver :presto
+    (is (= "UTC"
+           (metabase.driver/db-default-timezone :presto (mt/db))))))
 
 ;; Query cancellation test, needs careful coordination between the query thread, cancellation thread to ensure
 ;; everything works correctly together
 (deftest query-cancelation-test
-  (datasets/test-driver :presto
+  (mt/test-driver :presto
     (let [called-cancel-promise (atom nil)]
       (with-redefs [http/delete (fn [& _]
                                   (deliver @called-cancel-promise true))]
-        (is (= ::tu/success
-               (tu/call-with-paused-query
+        (is (= ::mt/success
+               (mt/call-with-paused-query
                 (fn [query-thunk called-query? called-cancel? pause-query]
                   (reset! called-cancel-promise called-cancel?)
                   (future
@@ -170,13 +171,13 @@
                       (query-thunk)))))))))))
 
 (deftest template-tag-timezone-test
-  (datasets/test-driver :presto
+  (mt/test-driver :presto
     (testing "Make sure date params work correctly when report timezones are set (#10487)"
-      (tu/with-temporary-setting-values [report-timezone "Asia/Hong_Kong"]
+      (mt/with-temporary-setting-values [report-timezone "Asia/Hong_Kong"]
         (is (= [["2014-08-02T00:00:00+08:00" "2014-08-02"]]
-               (qp.test/rows
+               (mt/rows
                  (qp/process-query
-                   {:database   (data/id)
+                   {:database   (mt/id)
                     :type       :native
                     :native     {:query         "SELECT {{date}}, cast({{date}} AS date)"
                                  :template-tags {:date {:name "date" :display_name "Date" :type "date"}}}

--- a/src/metabase/driver/util.clj
+++ b/src/metabase/driver/util.clj
@@ -38,7 +38,7 @@
     (try
       (can-connect-with-details? driver details-map :throw-exceptions)
       (catch Throwable e
-        (log/error (trs "Failed to connect to database: {0}" (.getMessage e)))
+        (log/error e (trs "Failed to connect to database"))
         false))))
 
 (defn report-timezone-if-supported

